### PR TITLE
Display map with markers of places near me that show description on click

### PR DESCRIPTION
**Pull request by haoyxc**
Add a map to display places around where I live I go to frequently as markers, markers show a description when clicked.

By default, the only thing in the container is a map with markers and the text appears when you click on a marker. 
image
(and if you're wondering, I think the reason the map is muted is because it says: You must enable Billing on the Google Cloud Project at https://console.cloud.google.com/project/_/billing/enable Learn more at https://developers.google.com/maps/gmp-get-started)

I wasn't sure of the best way to list all of the places and their descriptions. (One idea I had was to put them in a constants file? But I didn't know where that would go in the file tree -- was thinking webapp/constants -- but not entirely sure haha)

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -247,3 +247,45 @@ function updateHeaderToLogoutIfLoggedIn(isLoggedIn, url) {
...
+      document.getElementById('map'),
+      {center: {lat: 33.07196, lng: -96.771786}, zoom: 10});
+
+  locations.forEach(loc => {
+    const marker = new google.maps.Marker({
+      position: {lat: loc[1], lng: loc[2]}, 
+      map: map,
+      title: loc[0]
+    })
+    marker.addListener('click', showMarkerText)
```
Comment (fenghaolw) [inline portfolio/src/main/webapp/script.js]: add trailing ";"

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -247,3 +247,45 @@ function updateHeaderToLogoutIfLoggedIn(isLoggedIn, url) {
...
+  ]
+  const map = new google.maps.Map(
+      document.getElementById('map'),
+      {center: {lat: 33.07196, lng: -96.771786}, zoom: 10});
+
+  locations.forEach(loc => {
+    const marker = new google.maps.Marker({
+      position: {lat: loc[1], lng: loc[2]}, 
+      map: map,
+      title: loc[0]
```
Comment (fenghaolw) [inline portfolio/src/main/webapp/script.js:277]: add trailing ";"

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -247,3 +247,45 @@ function updateHeaderToLogoutIfLoggedIn(isLoggedIn, url) {
...
+ */
+function createMap() {
+  const locations = [
+    ["Trader Joe's", 33.029256, -96.793147], 
+    ["Plano West Senior High School", 33.0437, -96.8140], 
+    ["Sprouts Farmers Market", 33.069816, -96.773118]
+  ]
+  const map = new google.maps.Map(
+      document.getElementById('map'),
+      {center: {lat: 33.07196, lng: -96.771786}, zoom: 10});
```
Comment (fenghaolw) [inline portfolio/src/main/webapp/script.js]: nit: `);` should be in the next line

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -247,3 +247,45 @@ function updateHeaderToLogoutIfLoggedIn(isLoggedIn, url) {
...
+/**
+ * Create a map centered around Plano, Texas, where I'm from
+ * Each marker displays different text on click
+ */
+function createMap() {
+  const locations = [
+    ["Trader Joe's", 33.029256, -96.793147], 
+    ["Plano West Senior High School", 33.0437, -96.8140], 
+    ["Sprouts Farmers Market", 33.069816, -96.773118]
+  ]
```
Comment (fenghaolw) [inline portfolio/src/main/webapp/script.js]: add trailing `;`

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -247,3 +247,45 @@ function updateHeaderToLogoutIfLoggedIn(isLoggedIn, url) {
...
+      {center: {lat: 33.07196, lng: -96.771786}, zoom: 10});
+
+  locations.forEach(loc => {
+    const marker = new google.maps.Marker({
+      position: {lat: loc[1], lng: loc[2]}, 
+      map: map,
+      title: loc[0]
+    })
+    marker.addListener('click', showMarkerText)
+  })
```
Comment (fenghaolw) [inline portfolio/src/main/webapp/script.js]: add trailing `;`

**Review by fenghaolw — CHANGES_REQUESTED** [review]
Mostly minor linter issues

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (haoyxc) [reply portfolio/src/main/webapp/script.js:277]: this is in the next line right?
image

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -247,3 +247,46 @@ function updateHeaderToLogoutIfLoggedIn(isLoggedIn, url) {
...
+    marker.addListener('click', showMarkerText);
+  });
+}
+
+/**
+ * Show a description associated with a marker when a marker is clicked
+ * Clears the space first
+ */
+function showMarkerText(marker) {
+  const descriptions = {
```
Comment (EatingW) [inline portfolio/src/main/webapp/script.js]: Maybe pull this out to a global constant?

You can also define this constant in `createMap`, and pass this into `showMarkerText` via an arrow function.

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (fenghaolw) [reply portfolio/src/main/webapp/script.js:277]: right, sorry I mean `,` after `title: loc[0]`.

**Review by fenghaolw — APPROVED** [review]

**Review by EatingW — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (haoyxc) [reply portfolio/src/main/webapp/script.js:277]: if I fix it now and push, will it require you to approve again?

Comment (haoyxc) [reply portfolio/src/main/webapp/script.js:277]: actually I can just remember to fix it in the next PR haha

**Pull request merged**